### PR TITLE
The simplified template for sidebar menu

### LINKED ISSUE #3471: Autoactive cusrom item on navbar
look comments on config
```yml
sonata_admin:
    dashboard:
        groups:
            sonata.admin.group.content:
                label:           core.menu.routing
                label_catalogue: CoreBundle
                icon:            '<i class="fa fa-puzzle-piece"></i>'
                items:
                    - route:        application_session //if I pick this custom link on navbar, element not have class active
                      label:        'Session'
                    - application.admin.provider //all good, if i pick this item on meny, he highlighted and have active class
```

### PR DESCRIPTION
**Pull request by hason**
https://github.com/KnpLabs/KnpMenu/blob/master/doc/01-Basic-Menus.markdown#the-current-menu-item

Fixes #3471

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Is this BC?

**Comment by hason** [comment]
@Soullivaneuh I used a standard mechanism for detection an active item and removed obsolete code that duplicates this detection. Additionally, this also corrected css class for manually added items which are active. This PR breaks BC in some edge cases.

**Comment by Soullivaneuh** [comment]
I would take a look of this feature later.

Maybe you can explain a little bit more the goal of this PR and, if necessary, update the doc.

> This PR breaks BC in some edge cases.

Is it possible to make it BC? With deprecation system for sample.

For the moment, I'll put it on for the next major release milestone so.

cc @rande

**Comment by hason** [comment]
@Soullivaneuh I know of only one BC - https://github.com/sonata-project/SonataAdminBundle/pull/2855/files#diff-7aad62c164c47d66f90ce813f6f91c49R20 Should I preserve this functionality which was added in unrelated PR? The correct solution is use voters - https://github.com/KnpLabs/KnpMenu/blob/master/doc/01-Basic-Menus.markdown#the-current-menu-item

**Comment by hason** [comment]
Goals of this PR:
* simplify a template,
* use standard mechanisms from KnpMenu,
* fixed bug with wrong css class "current" for active custom items.

**Comment by StudioMaX** [comment]
+1

File: Resources/config/menu.xml [hunk 1]
```diff
@@ -5,12 +5,11 @@
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <services>
-        <service id="sonata.admin.menu_builder" class="Sonata\AdminBundle\Menu\MenuBuilder">
+        <service id="sonata.admin.menu_builder" class="Sonata\AdminBundle\Menu\MenuBuilder" public="false">
```
Comment (OskarStark) [inline Resources/config/menu.xml]: this looks like a BC break

File: Resources/views/standard_layout.html.twig [hunk 2]
```diff
@@ -215,9 +215,13 @@ file that was distributed with this source code.
...
                             {% block side_bar_before_nav %} {% endblock %}
                             {% block side_bar_nav %}
-                                {% if app.user and is_granted('ROLE_SONATA_ADMIN') %}
-                                    {{ knp_menu_render('sonata_admin_sidebar', {template: admin_pool.getTemplate('knp_menu_template')}) }}
-                                {% endif %}
+                                {{ knp_menu_render('sonata_admin_sidebar', {
+                                       currentClass: 'active',
+                                       ancestorClass: 'active',
+                                       branch_class: 'treeview',
+                                       template: admin_pool.getTemplate('knp_menu_template')}
```
Comment (OskarStark) [inline Resources/views/standard_layout.html.twig]: the ```}``` has to be on the next line i think

**Comment by OskarStark** [comment]
ping @hason

**Comment by hason** [comment]
@OskarStark Updated

**Comment by OskarStark** [comment]
Travis failed, please have a look

thank you @hason

**Comment by hason** [comment]
@OskarStark Fixed

**Comment by hason** [comment]
ping @OskarStark

**Comment by OskarStark** [comment]
Thank you.

@rande what do you think?

**Comment by muspelheim** [comment]
@OskarStark what about this?

**Comment by OskarStark** [comment]
i like this PR, @hason can you please rebase?

@core23 can you have a second look? thank you

**Comment by OskarStark** [comment]
EDIT: the only problem i see is that we remove a method... :(

**Comment by hason** [comment]
@OskarStark  rebased

**Comment by andrey1s** [comment]
@hason how to choose an active menu item?

**Comment by hason** [comment]
@andrey1s via voters in #3577 :wink:. This PR is uses only standard voters from KnpMenu.

File: Menu/MenuBuilder.php [hunk 3]
```diff
@@ -148,14 +137,4 @@ public function createSidebarMenu()
 
         return $event->getMenu();
     }
-
-    /**
-     * Sets the request the service.
-     *
-     * @param Request $request
-     */
-    public function setRequest(Request $request = null)
```
Comment (core23) [inline Menu/MenuBuilder.php]: Please keep the method for the next minor release and create a new PR for the next major version.

File: Resources/config/menu.xml [hunk 4]
```diff
@@ -10,7 +10,6 @@
             <argument type="service" id="knp_menu.factory" />
             <argument type="service" id="knp_menu.menu_provider" />
             <argument type="service" id="event_dispatcher" />
-            <call method="setRequest"><argument type="service" id="request" on-invalid="null" strict="false" /></call>
         </service>
```
Comment (core23) [inline Resources/config/menu.xml]: Same here

File: Menu/MenuBuilder.php (hunk 3 above)
Comment (hason) [inline Menu/MenuBuilder.php]: This method exists only in dev-master (2.4). It was introduced by me.

Comment (core23) [inline Menu/MenuBuilder.php]: Then I am fine with it :)

**Comment by OskarStark** [comment]
@hason can you provide a test that this works as expected, or is there already a testcase which passes?

**Comment by muspelheim** [comment]
I think this solution don't works.

@OskarStark, @hason I implement this logic to my project for test, and now if i picked custom link or other link from menu it's works, but if i go to EDIT any entity, menu loses active class.

**Comment by greg0ire** [comment]
@OskarStark @hason @core23 @Soullivaneuh : since @muspelheim showed this does not work yet, and no solution is proposed, this should be removed from the 2.4 milestone.

**Comment by core23** [comment]
Can you rebase this? Looks like it's partial rewritten with https://github.com/sonata-project/SonataAdminBundle/pull/3149

**Comment by Soullivaneuh** [comment]
@hason I added the `minor` label but `2.x` branch is coming very soon with a `2.4` release.

If this happen before this PR going to be merged, this will be a BC break for `getRequest` removal.

**Comment by greg0ire** [comment]
ping @hason

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**